clap_complete/src/dynamic/completer.rs

```diff
@@ -331,14 +346,54 @@ fn possible_values(a: &clap::Arg) -> Option<Vec<clap::builder::PossibleValue>> {
 ///
 /// Subcommand `rustup toolchain install` would be converted to
 /// `("install", "rustup toolchain install")`.
-fn subcommands(p: &clap::Command) -> Vec<(String, Option<StyledStr>)> {
+fn subcommands(p: &clap::Command) -> Vec<CompletionCandidate> {
     debug!("subcommands: name={}", p.get_name());
     debug!("subcommands: Has subcommands...{:?}", p.has_subcommands());
     p.get_subcommands()
         .flat_map(|sc| {
             sc.get_name_and_visible_aliases()
                 .into_iter()
-                .map(|s| (s.to_string(), sc.get_about().cloned()))
+                .map(|s| CompletionCandidate::new(s.to_string()).help(sc.get_about().cloned()))
         })
         .collect()
 }
+
+/// A completion candidate defination
+///
+/// This makes it easier to add more fields to completion candidate,
+/// rather than using `(OsString, Option<StyledStr>)` or `(String, Option<StyledStr>)` to represent a completion candidate
+#[derive(Default, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct CompletionCandidate {
+    /// Main completion candidate content
+    content: OsString,
+
+    /// Help message with a completion candidate
+    help: Option<StyledStr>,
+}
+
+impl CompletionCandidate {
+    /// Create a new completion candidate
+    pub fn new(content: impl Into<OsString>) -> Self {
+        let content = content.into();
+        Self {
+            content,
+            ..Default::default()
+        }
+    }
+
+    /// Set the help message of the completion candidate
+    pub fn help(mut self, help: Option<StyledStr>) -> Self {
+        self.help = help;
+        self
+    }
+
+    /// Get the content of the completion candidate
+    pub fn get_content(&self) -> &OsString {
+        &self.content
+    }
+
+    /// Get the help message of the completion candidate
+    pub fn get_help(&self) -> &Option<StyledStr> {
```

Comment:
While there is no borrowed type for `StyledStr`, `&` should be inside of `Option`.  So this should be `Option<&StyledStr>`

Reply:
Thanks for your review!